skfp/fingerprints/map.py

```diff
@@ -139,135 +144,124 @@ def __init__(
         )
         self.fp_size = fp_size
         self.radius = radius
-        self.variant = variant
+        self.include_duplicated_shingles = include_duplicated_shingles
+        self.counts = counts
 
     def _calculate_fingerprint(
         self, X: Sequence[Union[str, Mol]]
     ) -> Union[np.ndarray, csr_array]:
         X = ensure_mols(X)
         X = np.stack(
-            [self._calculate_single_mol_fingerprint(mol) for mol in X], dtype=int
+            [self._calculate_single_mol_fingerprint(mol) for mol in X],
+            dtype=np.uint32 if self.counts else np.uint8,
         )
 
-        if self.variant == "bit":
-            X = (X > 0).astype(np.uint8)
-        elif self.variant == "count":
-            X = X.astype(np.uint32)
-
         return csr_array(X) if self.sparse else np.array(X)
 
     def _calculate_single_mol_fingerprint(self, mol: Mol) -> np.ndarray:
         atoms_envs = self._get_atom_envs(mol)
-        shingles = self._get_atom_pair_shingles(mol, atoms_envs)
+        shinglings = self._get_atom_pair_shingles(mol, atoms_envs)
+
+        if self.counts:
+            folded = np.zeros(self.fp_size, dtype=np.uint32)
+            for shingling in shinglings:
+                hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+                folded[hashed % self.fp_size] += 1
+            return folded
+
+        folded = np.zeros(self.fp_size, dtype=np.uint8)
+        for shingling in shinglings:
+            hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
+            folded[hashed % self.fp_size] = 1
+        return folded
```

Comment:
I think that this can be deduplicated a bit:
```
dtype = np.uint32 if self.count else np.uint8
folded = np.zeros(self.fp_size, dtype=np.dtype)
for shingling in shinglings:
    hashed = struct.unpack("<I", sha256(shingling).digest()[:4])[0]
    idx = hashed % self.fp_size
    if self.count:
        folded[idx] += 1
    else:
        folded[idx] = 1
return folded
```

Reply:
Refactored as proposed